src/test-utils/dom/internal/dropdown-host.ts

```diff
@@ -154,6 +155,18 @@ export class DropdownContentWrapper extends ComponentWrapper {
     return dropdown.findOpenDropdown();
   }
 
+  /**
+   * Returns a selectable item (option wrapper) from the dropdown.
+   *
+   * @param optionIndex 1-based index of the option to select.
+   */
+  findSelectableItem(optionIndex: number): SelectableItemWrapper | null {
+    return this.findComponent(
+      `.${selectableStyles['selectable-item']}[data-test-index="${optionIndex}"]`,
+      SelectableItemWrapper
+    );
+  }
```

Comment:
There is already `.findDropdown().findOptionByValue(value)` and `.findDropdown().findOption(index)` utils.

Why does this code not use the same finders?

Reply:
The selectable item structure is the following:
```javascript
<SelectableItem>
  <Option />
  ...
  disabled state related content
</SelectableItem>
```

These methods find child component `Option`, while the new one finds the root element to have access to disabled state content. It was implemented this way to avoid props drilling to `Option` component

Reply:
Need to look at it from customer perspective.

Imagine you already have a bunch of tests using `findOptionByValue`. How much will they be surprised that `findOptionByValue().findDisabledReason()` does not work?

I think for this part we need to get back and revisit the API design to offer something more convenient to our users.

Reply:
I forgot to mention one important detail: `Option` component renders conditionally within `SelectableItem`, this way:
```javascript
{isParent ? (
  <span>{wrappedOption.label || wrappedOption.value}</span>
) : (
  <Option
    option={{ ...wrappedOption, disabled }}
    highlightedOption={highlighted}
    selectedOption={selected}
    highlightText={filteringValue}
  />
)}
```
If we put disabled content inside `Option` and make `findOptionByValue().findDisabledReason()` work, disabled reason won't be shown in the parent item, which is not a case. 

So we have 2 options here:
1. change `findOptionByValue` implementation to find `SelectableItem` instead of `Option`. Might make sense, but at the same time exposes additional risks, because it's a customer facing API
2. leave things as is, which is less risky. we don't know is customers will be surprised, even if they do, they can refer to testing documentation and solve this problem.

WDYT?

Reply:
I recommend putting yourself in shoes of customer and try writing some sample tests in a way how our customers use that (not for release in this PR, but just to get a sense of the API you are proposing).

When you have these tests we can look again at your PR.

Reply:
After some research came to realize that `findDisabledReason()` could be implemented within `OptionWrapper`, because they are requested via `createWrapper()` anyway (they are rendered through react portals)

So now test utils look the following way:

```javascript
expect(
  wrapper.findDropdown().findOption(1)!.findDisabledReason()!.getElement()
).toHaveTextContent('disabled reason');
```